src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/ItemManagerProvider.cs

```diff
@@ -23,6 +25,21 @@ internal class ItemManagerProvider : IAsyncCompletionItemManagerProvider
         public ItemManagerProvider(RecentItemsManager recentItemsManager)
             => _instance = new ItemManager(recentItemsManager);
 
-        public IAsyncCompletionItemManager GetOrCreate(ITextView textView) => _instance;
+        public IAsyncCompletionItemManager GetOrCreate(ITextView textView)
+        {
+            if (textView.TextBuffer.TryGetWorkspace(out var workspace))
+            {
+                var workspaceContextService = workspace.Services.GetRequiredService<IWorkspaceContextService>();
+
+                // If we're in a cloud environment context, we want to avoid returning a completion item manager.
```

Comment:
So if we're doing the Kind check in GoToDefinitionCommandHandler, can't we just do workspace.Kind == WorkspaceKind.CloudEnvironmentClientWorkspace and skip all the service stuff in the first place?

Reply:
I'm not sure - I think actually the gotodef command handler should use this.  As I'm removing C#_LSP, I think I'll be able to remove that workspace entirely, in which case this check will need to change to something else.  

Though perhaps that's premature thinking and it may be better to do the simple workspace kind check here, and defer the creation of this until I remove the workspace.